plugins/modules/aci_fabric_interface_policy_group.py

```diff
@@ -0,0 +1,359 @@
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        name=dict(type="str", aliases=["policy_group"]),
+        description=dict(type="str", aliases=["descr"]),
+        name_alias=dict(type="str"),
+        type=dict(type="str", aliases=["policy_group_type"], choices=["leaf", "spine"], required=True),
+        dwdm_policy=dict(type="str"),
+        link_level_policy=dict(type="str"),
+        link_flap_policy=dict(type="str"),
+        l3_interface_policy=dict(type="str"),
+        macsec_policy=dict(type="str"),
+        monitoring_policy=dict(type="str"),
+        transceiver_policy_tdn=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name"]],
+            ["state", "present", ["name"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+    policy_group_type = module.params.get("type")
+    dwdm_policy = module.params.get("dwdm_policy")
+    link_level_policy = module.params.get("link_level_policy")
+    link_flap_policy = module.params.get("link_flap_policy")
+    l3_interface_policy = module.params.get("l3_interface_policy")
+    macsec_policy = module.params.get("macsec_policy")
+    monitoring_policy = module.params.get("monitoring_policy")
+    transceiver_policy_tdn = module.params.get("transceiver_policy_tdn")
+
+    if policy_group_type == "leaf":
+        policy_group_class_name = "fabricLePortPGrp"
+        policy_group_class_rn = "leportgrp-{0}".format(name)
+    else:
+        policy_group_class_name = "fabricSpPortPGrp"
+        policy_group_class_rn = "spportgrp-{0}".format(name)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fabric",
+            aci_rn="fabric",
+        ),
+        subclass_1=dict(
+            aci_class="fabricFuncP",
+            aci_rn="funcprof",
+        ),
+        subclass_2=dict(
+            aci_class=policy_group_class_name,
+            aci_rn=policy_group_class_rn,
+            module_object=name,
+            target_filter={"name": name},
+        ),
+        child_classes=[
+            "fabricRsDwdmFabIfPol",
+            "fabricRsFIfPol",
+            "fabricRsFLinkFlapPol",
+            "fabricRsL3IfPol",
+            "fabricRsMacsecFabIfPol",
+            "fabricRsMonIfFabricPol",
+            "fabricRsOpticsFabIfPol",
+        ],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
```

Comment:
Should we handle the use case of deletion of child configuration when these are not provided and the object already exists?

Reply:
ok thanks, should we then document this is in the attribute description?

Reply:
Just to be consistent with other modules, ignore the documentation part. But it works when we provide an empty string to the child config policy group attributes.